src/api/controllers/db-controller.ts

```diff
@@ -504,28 +571,14 @@ async function getRosettaBlockTransactionsFromDataStore(opts: {
   const transactions: RosettaTransaction[] = [];
 
   for (const tx of txsQuery.result) {
-    let events: DbEvent[] = [];
-    if (blockQuery.result.block_height > 1) {
-      // only return events of blocks at height greater than 1
-      const eventsQuery = await opts.db.getTxEvents({
-        txId: tx.tx_id,
-        indexBlockHash: opts.indexBlockHash,
-        limit: 5000,
-        offset: 0,
-      });
-      events = eventsQuery.results;
-    }
-    const operations = await getOperations(tx, opts.db, minerRewards, events, unlockingEvents);
-    const txMemo = parseTransactionMemo(tx);
-    const rosettaTx: RosettaTransaction = {
-      transaction_identifier: { hash: tx.tx_id },
-      operations: operations,
-    };
-    if (txMemo) {
-      rosettaTx.metadata = {
-        memo: txMemo,
-      };
-    }
+    const rosettaTx = await parseRosettaTxDetail({
+      block_height: blockQuery.result.block_height,
+      indexBlockHash: opts.indexBlockHash,
+      tx,
+      db: opts.db,
+      minerRewards,
```

Comment:
This looks like you're going to be adding `minerRewards` and `unlockingEvents` operations to every single tx in the block, no? Shouldn't those appear only once in the coinbase tx?

Reply:
Nope, that is checked above [here](https://github.com/hirosystems/stacks-blockchain-api/pull/1072/files/99fdc200912888526e425f57a4d0bd132bab1c61#diff-36aedd0ad8fe395d5f8bde09febd4ebf512dee68243dde51da0863dd9807c172R529-R537) to fetch `minerRewards` and `unlockingEvents` only in case of `coinbase tx`. For other txs empty array is passed in the `getRosettaBlockTransactionsFromDataStore`. This functionality was replicated from [here](https://github.com/hirosystems/stacks-blockchain-api/blob/ee8060191269ddf637a7dae93a595ff67383a586/src/api/controllers/db-controller.ts#L518).